### ISSUE TITLE
Inconsistent NRGBA treatment between image/color and Ebitengine

### ISSUE THREAD

**Issue by tinne26** [hajimehoshi/ebiten#2749]
### Ebitengine Version

v2.5.9

### Operating System

- [ ] Windows
- [ ] macOS
- [ ] Linux
- [ ] FreeBSD
- [ ] OpenBSD
- [ ] Android
- [ ] iOS
- [ ] Nintendo Switch
- [ ] Xbox
- [ ] Web Browsers

### Go Version (`go version`)

go1.20.4

### What steps will reproduce the problem?

See https://gist.github.com/tinne26/8ac361ec83efcf655a6f55c6d0ec086e. To show one relevant excerpt:
```golang
// NRGBA test
for i := 0; i < 256; i++ {
	clr := color.NRGBA{ uint8(i), uint8(i), uint8(i), uint8(i) }
	img.Set(i, 0, clr)
	canvas.Set(i, 0, clr)
}
for i := 0; i < 256; i++ {
	a := img.At(i, 0).(color.RGBA)
	b := canvas.At(i, 0).(color.RGBA)
	if a != b {
		log.Printf("[NRGBA] image.RGBA.At(%d) = %v, canvas.At(%d) = %v", i, a, i, b)
		ok = false
	}
}
```

### What is the expected result?

Setting and getting colors on `image.RGBA` and `ebiten.Image` should be as consistent as possible. For `color.RGBA`, `color.Alpha` and others, this happens. For `image.NRGBA` and `image.NRGBA64`, it doesn't, and it looks like it's on the CPU side of conversions (https://github.com/hajimehoshi/ebiten/blob/v2.5.9/image.go#L930), not a GPU variance or precision error.

### What happens instead?

Results do not match.

### Anything else you feel useful to add?

I don't really know who's wrong or right here, but I couldn't find any previous issue for this; if there's a good reason for this behavior difference it very much seems to me that it should be mentioned somewhere.

**Comment by hajimehoshi** [comment]
I think this is just because of numerical errors. Apparently deltas are in the range of ±1.

```
...
2023/09/06 18:43:51 [NRGBA64] image.RGBA.At(248) = {242 242 242 248}, canvas.At(248) = {241 241 241 248}
2023/09/06 18:43:51 [NRGBA64] image.RGBA.At(249) = {244 244 244 249}, canvas.At(249) = {243 243 243 249}
2023/09/06 18:43:51 [NRGBA64] image.RGBA.At(250) = {246 246 246 250}, canvas.At(250) = {245 245 245 250}
2023/09/06 18:43:51 [NRGBA64] image.RGBA.At(251) = {248 248 248 251}, canvas.At(251) = {247 247 247 251}
2023/09/06 18:43:51 [NRGBA64] image.RGBA.At(252) = {250 250 250 252}, canvas.At(252) = {249 249 249 252}
```

**Comment by hajimehoshi** [comment]
So I don't have to (or cannot) fix this. What do you think?

**Comment by tinne26** [comment]
My whole point is that image/color and Ebitengine are doing different conversions for NRGBA colors. I don't see why such a difference should exist. I know the difference is small and it's not a big deal, but what's the reason for it? Without knowing the reason, we can't tell if this is a bug or not. Apparently it's image/color dividing by 256 instead of 257 for NRGBA initial downscaling, using `>> 8`... but which one is correct?

**Comment by tinne26** [comment]
Technically it should be fixable by using `color.RGBAModel.Convert(clr).(color.RGBA)` and then using those values instead, and that would be consistent, but I'm really intrigued by whether one of the two conversions is right or wrong.

**Comment by hajimehoshi** [comment]
> Technically it should be fixable by...

I don't think so. This depends on graphics driver's implementation, and this is out of control.

**Comment by tinne26** [comment]
No, this is happening on the CPU side computations:
> it looks like it's on the CPU side of conversions (https://github.com/hajimehoshi/ebiten/blob/v2.5.9/image.go#L930), not a GPU variance or precision error

I tested with this <https://github.com/tinne26/ebiten/blob/main/image.go#L965-L966> and all results match.

That tests may fail with this, or more conversions may happen in other places and other side effects... that I can't tell, it's above my paygrade, but the conversion differences, at least in Set(), are happening on the CPU side.

**Comment by tinne26** [comment]
It may still be best to leave this as it is due to the differences being very minor and back-compatibility being more relevant until v3, but this is not on the GPU drivers and it deserves some investigation regarding who has their conversion computations right.

**Comment by hajimehoshi** [comment]
Ok, but in general, ebiten.Image's At doesn't guarantee the precision. Even if your usage doesn't invoke GPU, this is just a coincident and an implementation detail. This might be changed in the future. So I don't think fixing this doesn't make sense.

**Comment by hajimehoshi** [comment]
I'm sorry but dividing by 256 seems correct... https://go.dev/play/p/eWnEgB145tA The values are distributed uniformally.

**Comment by hajimehoshi** [comment]
Though Ebitengine doesn't guarantee `At`'s precision, leaving a wrong logic is not reasonable. So I fixed it. Thank you, @tinne26!
